snappass/main.py

```diff
@@ -90,12 +90,53 @@ def request_is_valid(request):
     """
     return not SNEAKY_USER_AGENTS_RE.search(request.headers.get('User-Agent', ''))
 
+def not_found_api(error=None):
+    message = {
+            'status': 404,
+            'message': 'Not Found: ' + request.url,
+    }
+    resp = jsonify(message)
+    resp.status_code = 404
+
+    return resp
+
+def unsupported_media_type_api(error=None):
+    message = {
+            'status': 415,
+            'message': 'Unsupported Media Type',
+    }
+    resp = jsonify(message)
+    resp.status_code = 415
+
+    return resp
+
+def bad_request_api(error=None):
+    message = {
+            'status': 400,
+            'message': 'Bad Request',
+    }
+    resp = jsonify(message)
+    resp.status_code = 400
+
+    return resp
+
 
 @app.route('/', methods=['GET'])
 def index():
     return render_template('set_password.html')
 
 
+@app.route('/api', methods=['GET'])
+def index_api():
+    if NO_SSL:
+        base_url = request.url_root
+    else:
+        base_url = request.url_root.replace("http://", "https://")
```

Comment:
These four lines are repeated in multiple endpoint functions. Can we factor them into a common `make_base_url()` function?

Alternatively, we could just require [flask-sslify](https://github.com/kennethreitz/flask-sslify), which would move this concern out of snappass.